src/cdk/scrolling/viewport-ruler.ts

```diff
@@ -24,9 +24,8 @@ export const DEFAULT_RESIZE_TIME = 20;
  */
 @Injectable()
 export class ViewportRuler implements OnDestroy {
-
-  /** Cached document client rectangle. */
-  private _documentRect?: ClientRect;
+  /** Cached viewport dimensions. */
+  private _viewportSize: {width: number; height: number};
```

Comment:
Does reading the window width / height actually cost anything? Do we need to cache it at all?

Reply:
It does. See https://gist.github.com/paulirish/5d52fb081b3570c81e3a